### DISCUSSION TITLE
How to install raylib on Ubuntu 22 WSL2?

### DISCUSSION THREAD

**Discussion by Grave18** [raysan5/raylib#2897] [Q&A] [answered]
**In short. How tou run raylib on WSL2?**

On my laptop with Ubuntu 22 all works fine.

On my Ubuntu 22 WSl2 I configured XLaunch (X11 server for win10). Applications as Xcalc, VLC runs great.

My steps in terminal :

Prepearing:
sudo apt update
sudo apt install build-essential git
sudo apt install cmake
sudo apt install libasound2-dev mesa-common-dev libx11-dev libxrandr-dev libxi-dev xorg-dev libgl1-mesa-dev libglu1-mesa-dev

Building (embedded glfw):
cd ~/dev
git clone https://github.com/raysan5/raylib.git raylib
cd raylib/src/
make PLATFORM=PLATFORM_DESKTOP
sudo make install 

cd ../examples
make core/core_basic_window PLATFORM=PLATFORM_DESKTOP
./core/core_basic_window

Error:
![Screenshot_1](https://user-images.githubusercontent.com/5678288/215251041-0ade8840-1c6c-4751-8041-3da25c58601e.png)

Building (glfw3):
cd ~/dev
wget https://github.com/glfw/glfw/releases/download/3.2.1/glfw-3.2.1.zip
unzip glfw-3.2.1.zip
cd glfw-3.2.1
cmake -DBUILD_SHARED_LIBS=ON
sudo make install

cd raylib/src
sude make uninstall
make clean
make PLATFORM=PLATFORM_DESKTOP USE_EXTERNAL_GLFW=TRUE
sudo make install

cd ../examples
make core/core_basic_window PLATFORM=PLATFORM_DESKTOP USE_EXTERNAL_GLFW=TRUE
./core/core_basic_window

Error again:
![Screenshot_2](https://user-images.githubusercontent.com/5678288/215252035-3b14c9ba-fb08-479e-ba06-149dffa6efca.png)

**My suggesyions: figure out what is wrong and write wiki page about how to install raylib on WSL2.**

**Comment by chriscamacho** [comment, marked as answer]
you shouldn't need XLaunch, there is a requirement for additional windows graphics drivers (vGPU), and then for example apps like geany etc will just work.

https://learn.microsoft.com/en-us/windows/wsl/tutorials/gui-apps#prerequisites

**Reply by chriscamacho** [reply]
having not booted windows in a few months decided I'd better update it, while I was there double checked this, I can confirm there is NO need to run a Xwindows server, and using vGPU drivers allows raylib apps to work (specifically I tested the shaders_simple_lighting example)

**Reply by Grave18** [reply]
Indeed, Xlaunch was preventing the raylib window from launching. Thanks a lot. For about 3 days I couldn't figure out what was going on. I think that it can be mentioned somewhere.

**Reply by chriscamacho** [reply]
>  I think that it can be mentioned somewhere.
it literally is, in the link I provided...

**Reply by Grave18** [reply]
I mean maybe raylib wiki.

**Comment by chriscamacho** [comment]
thats for raylib stuff, the wsl2 instructions are on the microsoft site!

On Sun, 29 Jan 2023 at 07:30, Grave ***@***.***> wrote:

> I mean maybe raylib wiki.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/raysan5/raylib/discussions/2897#discussioncomment-4808007>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFO4SK35HPYRWGQSCL323DWUYMBBANCNFSM6AAAAAAUJPWUQA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 

blog bedroomcoders.co.uk <http://bedroomcoders.co.uk>
Free source code and tutorials!


Disclaimer:
By sending an email to ANY of my addresses you are agreeing that:

   1. I am by definition, "the intended recipient"

   2. All information in the email is mine to do with as I see fit and make
such financial profit, political mileage, or good joke as it lends itself
to. In particular, I may quote it where I please.

   3. I may take the contents as representing the views of your company.

   4. This overrides any disclaimer or statement of confidentiality that
may be included on your message.
